Extension/src/LanguageServer/cppbuildTaskProvider.ts

```diff
@@ -0,0 +1,320 @@
+        this.fileWatcher.onDidChange(() => this.doBuild());
+        this.fileWatcher.onDidCreate(() => this.doBuild());
+        this.fileWatcher.onDidDelete(() => this.doBuild());
+        // At this point we can start using the terminal.
+        this.writeEmitter.fire("Starting build...\r\n");
+        this.doBuild();
+    }
+
+    close(): void {
+        // The terminal has been closed. Shutdown the build.
+        if (this.fileWatcher) {
+            this.fileWatcher.dispose();
+        }
+    }
+
+    private async doBuild(): Promise<void> {
+        return new Promise<void>((resolve, reject) => {
+            // Do build.
+            const activeCommand: string = util.resolveVariables(this.command + " " + this.args.join(" "), this.AdditionalEnvironment);
+            if (this.options?.cwd) {
+                this.options.cwd = util.resolveVariables(this.options.cwd, this.AdditionalEnvironment);
+            }
+            cp.exec(activeCommand, this.options, (_error, stdout, stderr) => {
+                if (_error) {
+                    telemetry.logLanguageServerEvent("cppBuildTaskError", { "error": _error.message });
+                    this.writeEmitter.fire("Build finished with error:\r\n");
+                    this.writeEmitter.fire("\t" + _error.message + "\r\n");
+                    reject();
+                } else {
+                    this.writeEmitter.fire(stdout.toString());
+                    this.writeEmitter.fire("\r\nBuild finished successfully.\r\n");
+                    resolve();
+                }
+            });
+        }).finally (() => {
+            this.closeEmitter.fire();
+        });
+    }
+
+    private get AdditionalEnvironment(): { [key: string]: string | string[] } | undefined {
+
+        const editor: vscode.TextEditor | undefined = vscode.window.activeTextEditor;
+        if (!editor) {
+            return undefined;
+        }
+        const fileDir: vscode.WorkspaceFolder | undefined = vscode.workspace.getWorkspaceFolder(editor.document.uri);
+        if (!fileDir) {
+            return undefined;
+        }
+        const file: string = editor.document.fileName;
+        return {
+            "file": file,
+            "fileDirname": fileDir.uri.fsPath,
+            "fileBasenameNoExtension": path.parse(file).name,
+            "workspaceFolder": fileDir.uri.fsPath};
+    }
+}
+
+export function getRawTasksJson(): Promise<any> {
+    return new Promise<any>((resolve, reject) => {
+        const path: string | undefined = getTasksJsonPath();
+        if (!path) {
+            return resolve({});
+        }
+        fs.exists(path, exists => {
+            if (!exists) {
+                return resolve({});
+            }
+            let fileContents: string = fs.readFileSync(path).toString();
+            fileContents = fileContents.replace(/^\s*\/\/.*$/gm, ""); // Remove start of line // comments.
+            let rawTasks: any = {};
+            try {
```

Comment:
Would it be easy to switch to jsonc.parse instead? colorization.ts uses that. That would enable the comment removal code to be omitted. We might have another issue tracking that.